src/uu/sort/src/sort.rs

```diff
@@ -686,14 +686,21 @@ struct MergeableFile<'a> {
     lines: Box<dyn Iterator<Item = Line> + 'a>,
     current_line: Line,
     settings: &'a GlobalSettings,
+    file_index: usize,
 }
 
 // BinaryHeap depends on `Ord`. Note that we want to pop smallest items
 // from the heap first, and BinaryHeap.pop() returns the largest, so we
 // trick it into the right order by calling reverse() here.
 impl<'a> Ord for MergeableFile<'a> {
     fn cmp(&self, other: &MergeableFile) -> Ordering {
-        compare_by(&self.current_line, &other.current_line, self.settings).reverse()
+        let comparison = compare_by(&self.current_line, &other.current_line, self.settings);
+        if comparison == Ordering::Equal {
```

Comment:
please add a comment to explain the why

Reply:
FYI, I think semantics of unsorted inputs to -m is UB, but so long as we are using for an external sort (-S or -T), we need to stay on top of stable or unstable sorting.  

Re: -m, GNU says "merge already sorted files; do not sort."  BSD says "Merge only.  The input files are assumed to be pre-sorted.  If they are not sorted the output order is undefined."  As to what they actually do, it's anyone's guess.